src/main/java/org/folio/persist/IterationJobRepository.java

```diff
@@ -0,0 +1,26 @@
+package org.folio.persist;
+
+import static org.folio.rest.persist.PgUtil.postgresClient;
+
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import java.util.Map;
+import java.util.function.UnaryOperator;
+import org.folio.rest.jaxrs.model.IterationJob;
+
+public class IterationJobRepository extends AbstractRepository<IterationJob> {
+
+  public static final String TABLE_NAME = "iteration_job";
+
+  public IterationJobRepository(Context context, Map<String, String> okapiHeaders) {
+    super(postgresClient(context, okapiHeaders), TABLE_NAME, IterationJob.class);
+  }
+
+  public Future<IterationJob> fetchAndUpdate(String id, UnaryOperator<IterationJob> builder) {
```

Comment:
Why does a job need to be fetched in order to be updated?

Reply:
I've followed the pattern established by similar repository and method: [ReindexJobRepository.fetchAndUpdate()](https://github.com/folio-org/mod-inventory-storage/blob/11f09665e26bf47ebd841d47f860885cc278ff92/src/main/java/org/folio/persist/ReindexJobRepository.java#L18)

Why it's fetched first? Probably to get the latest representation of data and update it.